### PR TITLE
Carousel implementation

### PR DESCRIPTION
**Pull request by Ne-anna**
Created carousel organism which composed of carousel-content(text and images) and carousel buttons( left and right). 
React typescript.

### CODE REVIEW

**Review by ReinisZogots — APPROVED** [review]

File: src/Components/Organisms/carousel/carousel.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  carouselData: CarouselData;
+  carouselButtonData: CarouselButtons;
+}
+const Carousel = (props: CarouselProps) => {
+  const carousel = jsonFile.carousel;
+
+  const [slideIndex, setSlideIndex] = useState(1);
+  const [slideDirection, setSlideDirection] = useState("right");
+  const [prevSlide, setPrevSlide] = useState(carousel.length);
+  const [blockedAnimation, setBlockedAnimation] = useState(true);
```
Comment (ghost) [inline src/Components/Organisms/carousel/carousel.tsx]: provide types to useState

File: src/Components/Organisms/carousel/carousel.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+interface CarouselProps {
+  carouselData: CarouselData;
+  carouselButtonData: CarouselButtons;
+}
+const Carousel = (props: CarouselProps) => {
+  const carousel = jsonFile.carousel;
+
+  const [slideIndex, setSlideIndex] = useState(1);
+  const [slideDirection, setSlideDirection] = useState("right");
```
Comment (ghost) [inline src/Components/Organisms/carousel/carousel.tsx]: create enum for slide direction, use it

File: src/data.ts [hunk 3]
```diff
@@ -37,24 +40,35 @@ export interface SubMenu {
...
-export interface Carousel {
+export interface CarouselData {
+  carousel: CarouselItems[];
+  slideIndex?: number;
+  slideDirection?: string;
+  prevSlide?: number;
+  blockAnimation?: boolean;
+}
+
+export interface CarouselItems {
```
Comment (ghost) [inline src/data.ts]: should be singular

File: src/data.ts [hunk 4]
```diff
@@ -37,24 +40,35 @@ export interface SubMenu {
...
   title: string;
   subTitle: string;
   altTag: string;
   path: string;
 }
 
 export interface CarouselButtons {
```
Comment (ghost) [inline src/data.ts]: should be also singular

File: src/Components/Organisms/carousel/carousel.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  const [prevSlide, setPrevSlide] = useState(carousel.length);
+  const [blockedAnimation, setBlockedAnimation] = useState(true);
+
+  const nextSlide = () => {
```
Comment (ghost) [inline src/Components/Organisms/carousel/carousel.tsx]: optional, loadNextSlide could be more appropriate name
nextSlide sounds like a an object describing the next slide
same thing for previousSlide

**Review by ghost — APPROVED** [review]

**Pull request merged**